### ISSUE TITLE
RedundantIdentityWithTrue reports about `match (true) {...}`

### ISSUE THREAD

**Issue by lptn** [vimeo/psalm#7932]
sometimes `match (true) {...}` pattern is good for readability:

```php
$result = match (true) {
    $age >= 65 => 'senior',
    $age >= 25 => 'adult',
    $age >= 18 => 'young adult',
    default => 'kid',
};
```

I think `RedundantIdentityWithTrue` should not report about such cases. WDYT, guys?

https://psalm.dev/r/60e5238c86

**Comment by weirdan** [comment]
@lptn can you clarify what is the problem here?

**Comment by lptn** [comment]
@weirdan 
it's not an issue anymore. In previous Psalm versions it reported about `RedundantIdentityWithTrue`
